modules/javafx.controls/src/test/java/test/javafx/scene/control/ListViewTest.java

```diff
@@ -1121,7 +1121,7 @@ public void updateItem(String color, boolean empty) {
                 items.set(30, "yellow");
                 Platform.runLater(() -> {
                     Toolkit.getToolkit().firePulse();
-                    assertEquals(0, rt_35395_counter);
+                    assertTrue(rt_35395_counter < 7);
```

Comment:
I see that you have modified assertions to use "lesser than" some expected value. This may hide some incorrect test outcomes.
Along with "lesser than" assertion, do you think we should add a "greater than" assertion as well? This will have a range bounded expected value.
This is applicable for all assertion changes in this PR.

Reply:
The hard values have been changed a number of times, and I believe it is not really a good metric. 
What we want to ensure is that there is no functional regression and no performance penalty. The tests calculate the number of updateItem invocations and require those to be a strict number. With JDK-8089589, we fixed a number of issues that are related to the fact that the total size of the view (in case all cells are rendered with their preferred height) is not known. This is done by using an estimate of the total size. The estimate is 100% correct if we call updateItem for every item, but that would lead to a huge performance penalty in case the list contains a large amount of items with equal height.
Hence, there is a balance between minimizing the updateItem calls but still have a fair estimate of the total dimensions. Rather than requiring that the amount of calls should be a fixed number, I think it makes more sense to ensure that the amount of calls stays within reasonable boundaries, and is not growing exponentially when we add linearly more items, for example.

Reply:
> The hard values have been changed a number of times, and I believe it is not really a good metric.

I agree completely.

> Rather than requiring that the amount of calls should be a fixed number, I think it makes more sense to ensure that the amount of calls stays within reasonable boundaries, and is not growing exponentially when we add linearly more items, for example.

My point is exactly this. I see that as part of this PR, you have added the upper boundary (rather than a fixed number) for assertions. I am asking whether we need a lower boundary as well?

Reply:
And in this case, since the value used to be 0 before this fix, I think that's fine. For the other cases would it make sense to have a `> 0` check? Or are those cases for which 0 would be an OK value?